pkg/sql/drop_type.go

```diff
@@ -161,6 +161,11 @@ func (n *dropTypeNode) startExec(params runParams) error {
 func (p *planner) addTypeBackReference(
 	ctx context.Context, typeID, ref descpb.ID, jobDesc string,
 ) error {
+	// If it is a table implicit type, then there is no need to track such type reference.
```

Comment:
This is to support returning multiple columns as a tuple by using the table name. Previously this function is used by `create view` to track type references. And it would error out here if you use table name as type name in the view query. However, for `create function`, we need to allow it in the function signature to return a tuple type. So skipping here to avoid the `cannot modify table record type` (it's trying to get a mutable type descriptor with a table id). Cross validation would still error out for `create view` with similar message because it has a forward reference to the implicit type (actually a table). see the test change in `pkg/sql/logictest/testdata/logic_test/record`.  I was back and forward on should we add a reference to the table referenced as implicit type. Part of me is like the `types.T` has the OID and I think user is very likely to select from the actual table inside the function body, with which we would have cross reference to the table already (this maybe an unsafe assumption). But when I read you another comment, another part of me gave a thumbs up on tracking table references for it. Let me know how you feel...

Reply:
ok... I think the assumption is not safe..
in postgres you can do : 
```sql
create table t (a int primary key, b varchar);

create function f() returns setof t language sql as $$ select (1, 'hello') $$;

postgres=# select f();
     f
-----------
 (1,hello)
(1 row)
```

Reply:
alright, I removed this check from here to not change `create view` error on this. Added code for `create function` to track table reference if it's used as implicit type.